lib/management/patchConfiguration.js

```diff
@@ -18,63 +21,213 @@ function updateIngestionBuckets(locations, metadata, logger, cb) {
+                    supportsVersioning: true,
+                    connector: {
+                        hdclient: {
+                            bootstrap: l.details.bootstrapList,
+                        },
+                    },
+                };
+            }
+            break;
+        default:
+            log.info('unknown location type', { locationType:
+                l.locationType });
+            return;
+        }
+        location.sizeLimitGB = l.sizeLimitGB || null;
+        location.isTransient = Boolean(l.isTransient);
+        location.legacyAwsBehavior = Boolean(l.legacyAwsBehavior);
+        locations[location.name] = location;
+    });
+    return locations;
 }
 
-function patchConfiguration(overlayVersion, conf, metadata, ingestionEnabled,
-logger, cb) {
-    if (conf.version === undefined) {
+function updateLocations(conf) {
+    const locations = conf.locations;
+    const locationsWithReplicationBackend = Object.keys(locations)
+    // NOTE: In Orbit, we don't need to have Scality location in our
+    // destination bootstrapList config, since we do not replicate to
+    // any Scality Instance yet.
+    .filter(key => locations[key].locationType !== 'location-file-v1')
+    .reduce((obj, key) => {
+        /* eslint no-param-reassign:0 */
+        obj[key] = locations[key];
+        return obj;
+    }, {});
+    config.setBootstrapList(locationsWithReplicationBackend);
+
+    Object.keys(locations).forEach(locName => {
+        config.setIsTransientLocation(
+            locName, locations[locName].isTransient);
+    });
+}
+
+function patchConfiguration(
+    overlayVersion, patchConfig, metadata, ingestionEnabled, logger, cb
+) {
+    if (patchConfig === undefined ||
+        patchConfig.version === undefined ||
+        patchConfig.locations === undefined
+    ) {
         return process.nextTick(cb, null);
     }
-    if (conf && conf.locations) {
-        return async.waterfall([
-            next => {
-                // only update ingestion buckets list when ingestionEnabled
-                if (!ingestionEnabled) {
-                    return process.nextTick(next);
-                }
-                return updateIngestionBuckets(conf.locations, metadata, logger,
-                    next);
-            },
-            next => {
-                // only update bootstraplist on new overlay version
-                if (overlayVersion !== undefined &&
-                    overlayVersion >= conf.version) {
-                    return process.nextTick(next);
-                }
+
+    const conf = Object.assign({}, patchConfig);
+    return async.waterfall([
+        next => metadata.getObjectMD(
+            managementDatabaseName,
+            tokenConfigurationKey,
+            {},
+            logger,
+            next,
+        ),
+        (creds, next) => {
+            // only update bootstraplist on new overlay version
+            if (overlayVersion === undefined || conf.version > overlayVersion) {
```

Comment:
Regrouping the two conditional statements with the same conditions might be more readable?
```suggestion
            // only update locations and bootstrap list on new overlay version
            if (overlayVersion === undefined || conf.version > overlayVersion) {
                patchedLocations = patchLocations(conf.locations, creds, logger);
                conf.locations = patchedLocations;
                config.setPublicInstanceId(conf.instanceId);
                updateLocations(conf);
            } else {
                conf.locations = patchedLocations;
            }
```

Reply:
was trying to remove the else + conf.locations thing but yea it still looks better this way